src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -344,133 +344,312 @@ public class SqlMutationEngine : IMutationEngine
+                        // However, for PATCH and PUT requests, the primary key would be present in the request URL. For POST request, however, the primary key
+                        // would not be available in the URL and needs to be appened. So, the primary key of the newly created item which is stored in the primaryKeyRoute
+                        // is used to construct the Location Header.
+                        return ConstructCreatedResultResponse(mutationResultRow!.Columns, selectOperationResponse, primaryKeyRoute, isReadPermissionConfiguredForRole, isDatabasePolicyDefinedForReadAction);
                     }
 
-                    // Valid REST updates return OkObjectResult
-                    return OkMutationResponse(mutationResultRow.Columns);
+                    if (context.OperationType is EntityActionOperation.Update || context.OperationType is EntityActionOperation.UpdateIncremental)
+                    {
+                        return ConstructOkMutationResponse(mutationResultRow!.Columns, selectOperationResponse, isReadPermissionConfiguredForRole, isDatabasePolicyDefinedForReadAction);
+                    }
                 }
+
             }
 
             // if we have not yet returned, record is null
             return null;
         }
 
+        /// <summary>
+        /// Constructs a FindRequestContext from the Insert/Upsert RequestContext and the results of insert/upsert database operation.
+        /// For REST POST, PUT AND PATCH API reqeusts, when there are database policies defined for the read action,
+        /// a subsequent select query that honors the database policy is executed to fetch the results.
+        /// </summary>
+        /// <param name="context">Insert/Upsert Request context for the REST POST, PUT and PATCH request</param>
+        /// <param name="mutationResultRow">Result of the insert/upsert database operation</param>
+        /// <param name="roleName">Role with which the API request is executed</param>
+        /// <returns>Returns a FindRequestContext object constructed from the existing context and create/upsert operation results.</returns>
+        private FindRequestContext ConstructFindRequestContext(RestRequestContext context, DbResultSetRow mutationResultRow, string roleName)
+        {
+            FindRequestContext findRequestContext = new(entityName: context.EntityName, dbo: context.DatabaseObject, isList: false);
+
+            // PrimaryKeyValuePairs in the context is populated using the primary key values from the
+            // results of the insert/update database operation.
+            foreach (string primarykey in context.DatabaseObject.SourceDefinition.PrimaryKey)
+            {
+                if (_sqlMetadataProvider.TryGetBackingColumn(context.EntityName, primarykey, out string? backingColumnName))
+                {
+                    findRequestContext.PrimaryKeyValuePairs.Add(backingColumnName, value: mutationResultRow.Columns[primarykey]!);
```

Comment:
Thanks for catching this!

The `mutationResultRow` contains the mapped column names as keys. Also, I figured that it should be okay to set mapped column names in `findRequestContext.PrimarykeyValuePairs` because during the construction of `SqlQueryStructre`, the backing columns names are computed before further processing.   


I've updated the logic to index the `mutationResultRow` using exposed column name of each primary key in the table. 
`TryGetExposedColumnName()` populates either the mapped column name (if configured) or the original column name as is. It returns `true` when the column name exists. 
It returns false if the primary key does not exist in `EntityBackingColumnsToExposedNames`. But, that shouldn't happen as these info gets populated during startup 